geps/gep-1619.md

```diff
@@ -163,43 +163,299 @@ Java application servers such as Tomcat and Jetty, were the first to standardize
+
+On the other hand, session persistence also augments routing. An xRoute, as defined by Gateway API, specifies how
+requests from a Gateway listener to an API object (such as a service) are routed. In routing scenarios involving traffic
+splitting, session persistence augments the routing behavior. When a persistent session is established and traffic
+splitting is configured across services, the persistence to a single backend should be maintained across services. In
+other words, the persistent routing behavior should disregard the traffic splitting configuration and remain consistent.
+
+**What are the practical implications of attaching session persistence to an existing API object?**
+
+One practical implication of attaching session persistence configuration to services is its reusability across multiple
+routes. This reusability can be advantageous as it eliminates the need for repetitive session persistence configuration
+for each individual route. However, on the flip side, it also mandates the inclusion of session persistence
+configuration for every route that utilizes the service.
+
+Conversely, when attaching session persistence configuration to xRoutes, it becomes possible to establish a clear
+relationship that it augments routing behavior, particularly in scenarios involving traffic splitting. However, as
+mentioned earlier, session persistence would need to be individually configured for each specific route.
+
+**How are existing implementation APIs designed?**
+
+Referring to our [Implementations](#Implementations) tables on session persistence, the majority of Gateway API
+implementations designed session persistence in their APIs to be attached to a service or backends. This should be
+considered cautiously, as making associations to Gateway API's notion of Gateway, Route, and Service to other
+implementation's objects is hard to directly translate. The idea of a route in Gateway API is often not the same as a
+route in any given implementation.
+
+Taking Istio as an example, it offers an attachment point of session persistence to a service through a [`DestinationRule`](https://istio.io/latest/docs/reference/config/networking/destination-rule/#DestinationRule).
+`DestinationRule` defines policies that apply to traffic intended for a service after routing has occurred. Within
+`DestinationRule`, Istio couples the selection of load balancing algorithms with session persistence via their [`LoadBalancerSettings`](https://istio.io/latest/docs/reference/config/networking/destination-rule/#LoadBalancerSettings).
+When designing our API structure, it is important to consider the balance between incorporating additional load
+balancing policies coupled with session persistence, rather than configuring session persistence independently.
+
+On the other hand, rather than being attached to a service, Envoy configures session persistence at the route-level
```

Comment:
> the above mentioned Envoy design is based on an "HTTP Filter" that is a route level configuration (and not service or "cluster" level configuration) and that enables the data plane entities to implement session persistence across services (clusters). It will be good to take that into account

@sanjaypujare resolved here.